odxtools/utils.py

```diff
@@ -2,13 +2,40 @@
 import dataclasses
 import re
 from typing import TYPE_CHECKING, Any, Dict, Optional
+from xml.etree import ElementTree
+
+from .exceptions import odxraise
 
 if TYPE_CHECKING:
     from .database import Database
     from .diaglayers.diaglayer import DiagLayer
     from .snrefcontext import SnRefContext
 
 
+def read_hex_binary(et_element: Optional[ElementTree.Element]) -> Optional[int]:
+    """Convert the contents of an xsd:hexBinary to a `bytes` object"""
+    if et_element is None:
+        return None
+
+    if (bytes_str := et_element.text) is None:
+        odxraise(f"Element {et_element.tag} has no content")
```

Comment:
Empty text is valid, it means zero length byte array

Reply:
you can try it with an xsd validator, `<BIT-MASK />` is a valid `xsd:hexBinary`, most libraries don't differentiate inline termination from empty string, and consider them the same thing.

Reply:
good points. fixed, thanks: https://github.com/mercedes-benz/odxtools/pull/404/commits/866b5434e670fdc7623e0efc8e248461e80ddb8d